File: examples/linear.py
from argparse import ArgumentParser

from franky import Affine, CartesianMotion, Robot, ReferenceType


if __name__ == "__main__":
    parser = ArgumentParser()
    parser.add_argument("--host", default="172.16.0.2", help="FCI IP of the robot")
    args = parser.parse_args()

    # Connect to the robot
    robot = Robot(args.host)
    robot.relative_dynamics_factor = 0.05
    robot.recover_from_errors()

    # Reduce the acceleration and velocity dynamic
    robot.relative_dynamics_factor = 0.15

    # Define and move forwards
    target = Affine([0.0, 0.2, 0.0])
    motion_forward = CartesianMotion(target, ReferenceType.Relative)
    robot.move(motion_forward)

    # And move backwards using the inverse motion
    motion_backward = CartesianMotion(target.inverse)
    robot.move(motion_backward)

File: examples/linear.cpp
#include <franky.hpp>

#include "util.cpp"


using namespace franky;


int main(int argc, char *argv[]) {
  if (argc != 2) {
    std::cerr << "Usage: " << argv[0] << " <robot-hostname>" << std::endl;
    return -1;
  }

  // Connect to the robot
  Robot robot(argv[1]);
  robot.automaticErrorRecovery();

  // Reduce the acceleration and velocity dynamic
  robot.setRelativeDynamicsFactor(0.15);

  // Define and move forwards
  auto way = mk_affine(0.0, 0.05, 0.0);
  auto motion_forward = std::make_shared<CartesianMotion>(RobotPose(way), ReferenceType::Relative);
  robot.move(motion_forward);

  // And move backwards using the inverse motion
  auto motion_backward = std::make_shared<CartesianMotion>(RobotPose(way.inverse()), ReferenceType::Relative);
  robot.move(motion_backward);

  return 0;
}
